src/components/ShareTrajectoryModal/index.tsx

```diff
@@ -93,18 +81,218 @@ const ShareTrajectoryModal = ({
+        } else if (setting === "height") {
+            roundedLimitedInput = numericInput >= 2160 ? 2160 : numericInput;
         }
+
+        setEmbedSettings({ ...embedSettings, [setting]: roundedLimitedInput });
     };
 
+    const EmbedSnippetPanel: JSX.Element = (
+        <VerticalFlexbox gap={8}>
+            <div className={styles.embedHeader}>
+                <h4>Embed &lt;/&gt;</h4>
+                <button
+                    className={styles.settingsButton}
+                    onClick={() => {
+                        setShowEmbedSettingsPanel(!showEmbedSettingsPanel);
+                    }}
+                >
+                    Show settings {DownArrow}
+                </button>
+            </div>
+            {showEmbedSettingsPanel && (
+                <div className={styles.embedSettings}>
+                    <VerticalFlexbox gap={4}>
+                        <p className={styles.accentText}> Size </p>
+                        <VerticalFlexbox gap={24}>
+                            <div className={styles.proportionSettings}>
+                                {" "}
+                                <Input
+                                    disabled={
+                                        !embedSettings.userProvidedProportions
+                                    }
+                                    defaultValue={embedSettings.height}
+                                    className={styles.numberInputs}
+                                    onChange={(e) => {
+                                        handleEmbeddedSizeInputs(
+                                            "height",
+                                            e.target.value
+                                        );
+                                    }}
+                                />
+                                <p> x </p>
+                                <Input
+                                    disabled={
+                                        !embedSettings.userProvidedProportions
+                                    }
+                                    value={embedSettings.width}
+                                    className={styles.numberInputs}
+                                    onChange={(e) => {
+                                        handleEmbeddedSizeInputs(
+                                            "width",
+                                            e.target.value
+                                        );
+                                    }}
+                                />
+                                <p> pixels </p>
+                                <div className={styles.constrainProportions}>
+                                    <Checkbox
+                                        onChange={() => {
+                                            setEmbedSettings({
+                                                ...embedSettings,
```

Comment:
constrain proportions shouldn't disable/enable the inputs. it means you keep the ratio of height to width constant. ie, if they change the width and it's checked, it will scale the height appropriately